_docs/tasks/traffic-management-v1alpha3/ingress.md

```diff
@@ -0,0 +1,621 @@
+    ClusterRoles and ClusterRoleBindings according to those recorded copies.    
+   
+
+## Using Istio Routing Rules with Ingress
+
+Istio's routing rules can be used to achieve a greater degree of control
+when routing requests to backend services. For example, the following
+route rule sets a 4s timeout for all calls to the httpbin service on the
+/delay URL.
+
+```bash
+cat <<EOF | istioctl create -f -
+apiVersion: config.istio.io/v1alpha2
+kind: RouteRule
+metadata:
+  name: status-route
+spec:
+  destination:
+    name: httpbin
+  match:
+    # Optionally limit this rule to istio ingress pods only
+    source:
+      name: istio-ingress
+      labels:
+        istio: ingress
+    request:
+      headers:
+        uri:
+          prefix: /delay/ #must match the path specified in ingress spec
+              # if using prefix paths (/delay/.*), omit the .*.
+              # if using exact match, use exact: /status
+  route:
+  - weight: 100
+  httpReqTimeout:
+    simpleTimeout:
+      timeout: 4s
+EOF
+```
+
+If you were to make a call to the ingress with the URL
+`http://$INGRESS_HOST/delay/10`, you will find that the call returns in 4s
+instead of the expected 10s delay.
+
+You can use other features of the route rules such as redirects, rewrites,
+routing to multiple versions, regular expression based match in HTTP
+headers, websocket upgrades, timeouts, retries, etc. Please refer to the
+[routing rules]({{home}}/docs/reference/config/istio.routing.v1alpha1.html)
+for more details.
+
+> Note 1: Fault injection does not work at the Ingress
+
+> Note 2: When matching requests in the routing rule, use the same exact
+> path or prefix as the one used in the Ingress specification.
+
+## Understanding ingresses
+
+Ingresses provide gateways for external traffic to enter the Istio service
+mesh and make the traffic management and policy features of Istio available
+for edge services.
+
+The servicePort field in the Ingress specification can take a port number
+(integer) or a name. The port name must follow the Istio port naming
+conventions (e.g., `grpc-*`, `http2-*`, `http-*`, etc.) in order to
+function properly. The name used must match the port name in the backend
+service declaration.
+
+In the preceding steps we created a service inside the Istio service mesh
+and showed how to expose both HTTP and HTTPS endpoints of the service to
+external traffic. We also showed how to control the ingress traffic using
+an Istio route rule.
+
+## Cleanup
+
+1. Remove the secret and Ingress Resource definitions.
+    
+   ```bash
+   kubectl delete ingress simple-ingress secure-ingress 
+   kubectl delete -n istio-system secret istio-ingress-certs
+   ```
+
+1. Shutdown the [httpbin](https://github.com/istio/istio/tree/master/samples/httpbin) service.
+
+   ```bash
+   kubectl delete -f samples/httpbin/httpbin.yaml
+   ```
+
+
+## What's next
+
+* Learn more about [Ingress Resources](https://kubernetes.io/docs/concepts/services-networking/ingress/).
+
```

Comment:
we can skip this doc and wait for gateway to land

Reply:
I think it's OK to keep the part of ingress that does not require combining route rules. We _do_ want basic ingress to continue to work, since it's crucial part of UX for the majority of k8s customers.

Reply:
I was thinking that we could add another section in this task to talk about gateway (when it's working), so the "Ingress Task" will show how to do ingress both ways. For now, I guess it's best to just leave it as is, and enhance it in a later PR.

Reply:
I'm going to remove the ingress task from this PR for now, and then add a new combined ingress/gateway task in a followup PR.